maat-court-data-api/src/main/java/gov/uk/courtdata/repository/HardshipReviewRepository.java

```diff
@@ -15,4 +17,8 @@ public interface HardshipReviewRepository extends JpaRepository<HardshipReviewEn
 
     @Query(value = "SELECT hr FROM HardshipReviewEntity hr WHERE hr.repId = :repId AND hr.replaced = 'N'")
     HardshipReviewEntity findByRepId(@Param("repId") Integer repId);
+
+    @Query(value = "SELECT hr.* FROM TOGDATA.HARDSHIP_REVIEWS hr, TOGDATA.HARDSHIP_REVIEW_DETAILS hrd " +
+            "WHERE HR.ID = HRD.HARD_ID and HRD.HRDT_TYPE = :detailType AND HR.REP_ID = :repId", nativeQuery = true)
+    Optional<HardshipReviewEntity> findByDetailType(@Param("detailType") String detailType, @Param("repId") Integer repId);
```

Comment:
I'm not sure if it is the lack of uniqueness of the search criteria, or an issue with the join, but the above query can return multiple records (see below).

I would recommend you amend the query and test it against the dev DB and verify constraints to ensure it can only return a maximum of 1 record.
Or otherwise change the method signature in the HardshipReviewRepository to return type `Optional<List<HardshipReviewEntity>>`

```
SELECT
    count(*)
FROM
    togdata.hardship_reviews        hr,
    togdata.hardship_review_details hrd
WHERE
        hr.id = hrd.hard_id
    AND hrd.hrdt_type = 'EXPENDITURE'
    AND hr.rep_id = 4351623;

"COUNT(*)"
11
```

Reply:
Updated the method signature to accommodate multiple records.